### PR TITLE
simplified conan test command

### PR DESCRIPTION
**Pull request by memsharded**

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -134,39 +134,15 @@ def test(self, *args):
         parser = argparse.ArgumentParser(description=self.test.__doc__, prog="conan test")
         parser.add_argument("path", help='path to a recipe (conanfile.py), e.g., conan test '
                                          'pkg/version@user/channel ')
```
Comment (lasote) [inline conans/client/command.py]: Change this comment. path is not a reference anymore.

**Comment by lasote** [comment]
Add a doc issue to update the command reference and examples

**Comment by lasote** [comment]
Also, take a look to the last comment of @solvingj, it is not well explained/confusing which conanfile is talking about (because it's common to have the recipe one and the test_package one). Given that it's not mandatory to have that structure to run a `conan test` we might add a comment like: `if you have a test_package folder, the path must point to it`

File: conans/client/command.py (hunk 1 above)
Comment (lasote) [reply conans/client/command.py]: See my comment below.

File: conans/client/conan_api.py [hunk 2]
```diff
@@ -696,19 +695,6 @@ def export_alias(self, reference, target_reference):
 Conan = ConanAPIV1
 
 
-def _check_query_parameter_and_get_reference(query, pattern):
```
Comment (memsharded) [inline conans/client/conan_api.py]: dead code!

File: conans/client/conan_api.py [hunk 3]
```diff
@@ -738,19 +724,6 @@ def get_conan_runner():
     return runner
 
 
-def _get_reference(ref, cwd=None):
```
Comment (memsharded) [inline conans/client/conan_api.py]: dead code

**Comment by solvingj** [comment]
@lasote regarding "It's common to have the recipe one and the test_package one. "  

`test_package` is assumed the default by the rest of conan. For example, `conan create` assumes `test_package`, despite it not being mandatory. Is it not reasonable for `conan test` to make same assumptions and make it use `test_package` as the default folder?

**Comment by lasote** [comment]
`conan create` do not require the `test_package` anymore. You can create a binary package without it, that's why the new `conan test` follow the same decoupled behavior. You could test even a remote recipe just having a conanfile.py with a `test` method.

**Comment by solvingj** [comment]
As you said, `conan create` will automatically "use"  `test_package` folder if present (I.E. it does not need to be provided as an argument, it's assumed, but can be overridden with an argument).  This is the behavior I'm suggesting for `conan test`

**Comment by lasote** [comment]
I discussed it some days ago with @memsharded and I do not agree. Explicit better than implicit, python zen. `conan test` doesn't need to be coupled to any special directory structure.

**Comment by solvingj** [comment]
ok, thanks for consideration.

**Pull request merged**